src/EFCore.Design/Extensions/ScaffoldingModelExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.EntityFrameworkCore;
+
+/// <summary>
+/// Design-time model extensions.
+/// </summary>
+public static class ScaffoldingModelExtensions
+{
+    /// <summary>
+    /// Check whether an entity type could be considered a many-to-many join entity type.
+    /// </summary>
+    /// <param name="entityType">The entity type to check.</param>
+    /// <returns><see langword="true"/> if the entity type could be considered a join entity type.</returns>
+    public static bool IsManyToManyJoinEntityType(this IEntityType entityType)
```

Comment:
If this going to be a public extension it needs to have a more accurate name
```suggestion
    public static bool IsSimpleManyToManyJoinEntityType(this IEntityType entityType)
```